### DISCUSSION TITLE
`DxcTooltip` behavior for disabled content

### DISCUSSION THREAD

**Discussion by Mil4n0r** [dxc-technology/halstack-react#1983] [General]
The recently added `DxcTooltip` does not have a specific implementation on how to deal with disabled content.

I would like to know if there are any specific scenarios where it would be interesting to have a tooltip in a disabled component. That is not the default behavior with the HTML tooltip but we would have to program some mechanism to prevent it from triggering with our custom tooltip.

In my opinion, the most normal approach would be to conditionally render the tooltip from within the component that triggers the tooltip itself.

This is easier to see in code:

```
// ExampleComponent.tsx

// Previous:
return(
  <DxcTooltip label="Tooltip Test" position="bottom">
    <ExampleComponent>
      ...
    </ExampleComponent>
  </DxcTooltip>,
);
// New
return(!disabled ? 
  (<DxcTooltip label="Tooltip Test" position="bottom">
    <ExampleComponent>
      ...
    </ExampleComponent>
  </DxcTooltip>)
:
  <ExampleComponent>
    ...
  </ExampleComponent>
);
```

The drawbacks of this solution is that it would lead to repeat this condition for all the components that can be disabled.

**Comment by Jialecl** [comment]
In my opinion tooltips should not appear in disabled components, but it seems that there is no agreement on how tooltips should work with a disabled component.

There is also another possible solution, which is adding a new prop `disable` to the tooltip.

This comment explains pretty well the situeation:
https://github.com/mui/material-ui/issues/33182#issuecomment-1511626242
